### ISSUE TITLE
Stop ignoring dependency changes in the Cargo.lock up-to-date check 

### ISSUE THREAD

**Issue by teor2345** [ZcashFoundation/zebra#3388]
## Motivation

In PR #3363, we changed the `zcash_script` version in Cargo.toml, but we didn't update Cargo.lock.

This dependency bug should have been caught by the Cargo.lock CI check:
https://github.com/ZcashFoundation/zebra/blob/ebd94b2e60c01f74042db4a5899b294659d5f207/.github/workflows/ci.yml#L305-L310

But it seems like the dependency check isn't working.

(It is also possible that this check doesn't work on git versions. But that would be surprising undocumented behaviour.)

### Scheduling

This is a medium-priority security issue.

The dependency check ensures that Zebra is built with the correct versions of each crate it depends on. If we aren't doing this check, we could accidentally build Zebra with dependencies that are vulnerable to security issues.

## Command Reference

> requires that the Cargo.lock file is up-to-date. If the lock file is missing, or it needs to be updated, Cargo will exit with an error.
>
> These may be used in environments where you want to assert that the Cargo.lock file is up-to-date (such as a CI build)

https://doc.rust-lang.org/cargo/commands/cargo-check.html#manifest-options

**Comment by teor2345** [comment]
Here's what I see when I run that command on the `main` branch (with PR #3363, but without PR #3387):
```sh
$ cargo check --locked --all-features --all-targets 
    Updating git repository `https://github.com/ZcashFoundation/zcash_script.git`
    Updating crates.io index
error: the lock file /home/dev/zebra/Cargo.lock needs to be updated but --locked was passed to prevent this
If you want to try to generate the lock file without accessing the network, remove the --locked flag and use --offline instead.
Exit 101
```

I'm using:
```sh
$ cargo --version
cargo 1.60.0-nightly (95bb3c92b 2022-01-18)
```

**Comment by teor2345** [comment]
Here's what happened in CI on PR #3363:

```
Run actions-rs/cargo@v1.0.3
  with:
    command: check
    args: --locked --all-features --all-targets
    use-cross: false
  env:
    CARGO_INCREMENTAL: 0
    RUST_BACKTRACE: full
    CACHE_ON_FAILURE: false
/home/runner/.cargo/bin/cargo check --locked --all-features --all-targets
   Compiling zebra-chain v1.0.0-beta.3 (/home/runner/work/zebra/zebra/zebra-chain)
   Compiling zebra-state v1.0.0-beta.3 (/home/runner/work/zebra/zebra/zebra-state)
    Checking zebra-client v1.0.0-beta.0 (/home/runner/work/zebra/zebra/zebra-client)
    Checking zebra-rpc v1.0.0-beta.0 (/home/runner/work/zebra/zebra/zebra-rpc)
    Checking tower-fallback v0.2.15 (/home/runner/work/zebra/zebra/tower-fallback)
    Checking tower-batch v0.2.19 (/home/runner/work/zebra/zebra/tower-batch)
   Compiling zebrad v1.0.0-beta.3 (/home/runner/work/zebra/zebra/zebrad)
    Checking zebra-test v1.0.0-beta.3 (/home/runner/work/zebra/zebra/zebra-test)
    Checking zebra-script v1.0.0-beta.3 (/home/runner/work/zebra/zebra/zebra-script)
    Checking zebra-network v1.0.0-beta.3 (/home/runner/work/zebra/zebra/zebra-network)
    Checking zebra-consensus v1.0.0-beta.3 (/home/runner/work/zebra/zebra/zebra-consensus)
    Checking zebra-utils v1.0.0-beta.3 (/home/runner/work/zebra/zebra/zebra-utils)
    Finished dev [unoptimized + debuginfo] target(s) in 34.40s
```

It was using:
```
stable-x86_64-unknown-linux-gnu updated - rustc 1.58.0 (02072b482 2022-01-11) (from rustc 1.57.0 (f1edd0429 2021-11-29))
```

https://github.com/ZcashFoundation/zebra/runs/4870820023?check_suite_focus=true

**Comment by teor2345** [comment]
This should be fixed by PR #3377.

I think what was happening here is that we ran clippy, which updated the lock file, then we ran the checks.
